CommunicationLayer/MessagingFramework/MotorDetailsMessage.cpp

```diff
@@ -0,0 +1,153 @@
+    const int VOLTAGE_RAIL_SUPPLY_19_V = 41;
+    const int HEAT_SINK_TEMPERATURE = 45;
+    const int MOTOR_TEMPTERATURE = 49;
+    const int DSP_BOARD_TEMPERATURE = 53;
+    const int DC_BUS_AMP_HOURS = 57;
+    const int ODOMETER = 61;
+    const int SLIP_SPEED = 65;
+}
+
+MotorDetailsMessage::MotorDetailsMessage(const QByteArray& messageData, unsigned char motorNumber)
+: messageData_(messageData)
+, motorNumber_(motorNumber)
+{
+}
+
+float MotorDetailsMessage::phaseCCurrent() const
+{
+    return getFloat(messageData_, PHASE_CCURRENT);
+}
+
+float MotorDetailsMessage::phaseBCurrent() const
+{
+    return getFloat(messageData_, PHASE_BCURRENT);
+}
+
+float MotorDetailsMessage::motorVoltageReal() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_REAL);
+}
+
+float MotorDetailsMessage::motorVoltageImaginary() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_IMAGINARY);
+}
+
+float MotorDetailsMessage::motorCurrentReal() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_REAL);
+}
+
+float MotorDetailsMessage::motorCurrentImaginary() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_IMAGINARY);
+}
+
+float MotorDetailsMessage::backEmfReal() const
+{
+    return getFloat(messageData_, BACK_EMF_REAL);
+}
+
+float MotorDetailsMessage::backEmfImaginary() const
+{
+    return getFloat(messageData_, BACK_EMF_IMAGINARY);
+}
+
+float MotorDetailsMessage::voltageRailSuppply15V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPPLY_15_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply33V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_33_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply19V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_19_V);
+}
+
+float MotorDetailsMessage::heatSinkTemperature() const
+{
+    return getFloat(messageData_, HEAT_SINK_TEMPERATURE);
+}
+
+float MotorDetailsMessage::motorTempterature() const
+{
+    return getFloat(messageData_, MOTOR_TEMPTERATURE);
+}
+
+float MotorDetailsMessage::dspBoardTemperature() const
+{
+    return getFloat(messageData_, DSP_BOARD_TEMPERATURE);
+}
+
+float MotorDetailsMessage::dcBusAmpHours() const
+{
+    return getFloat(messageData_, DC_BUS_AMP_HOURS);
+}
+
+float MotorDetailsMessage::odometer() const
+{
+    return getFloat(messageData_, ODOMETER);
+}
+
+float MotorDetailsMessage::slipSpeed() const
+{
+    return getFloat(messageData_, SLIP_SPEED);
+}
+
+QString MotorDetailsMessage::toString() const
+{
+    QString messageString;
+    if (motorNumber_ == 0x0) {
+        messageString += QString::number(MessageDefines::Motor0Details) + ", ";
+    }
+    else {
+        // For the sake of not halting programs execution, treat all other cases as motor one
+        if (motorNumber != 0x1) {
+            // Log that this was used incorrectly
+            qDebug() << "Invalid motor number, defaulting to motor 1";
+        }
+        messageString += QString::number(MessageDefines::Motor1Details) + ", ";
+    }
```

Comment:
@enochtsang  

In this case there are only two motors defined in MessageDefines (i.e. Motor 0 and Motor 1). So I'm not sure what the normal flow of instruction would be if given an incorrect motor number as it would not be defined. I could implement so it is calculating the package ID via the following
```
messageString += QString::number(MessageDefines::Motor0Details + motorNumber_) + ", ";
``` 

But I personally feel this would be far more difficult to debug. I could do this in conjunction with a debug statement, but on the off chance it didn't cause problems it would suck if faulty code somehow made it's way into the car. 

In any case let me know, I'll go with whatever you decide